src/push_relabel.jl

```diff
@@ -169,6 +169,21 @@ function relabel! end
     nothing
 end
 
+"""
+    is_zero(value)
+
+Test if the value is equal to zero. It handles floating point errors.
+"""
+function is_zero end
```

Comment:
This function is not necessary, you can use `isapprox(x, 0)` or `x ≈ 0` in the code

Reply:
The default isapprox is not well suited for our problem. If `isapprox(0.2+0.1, 0.3)` is returning `true`, `isapprox(0.2+0.1-0.3, 0.0)` is returning `false`. In fact,  `x ≈ 0` is equivalent to `x == 0`.
As we compare to 0, rtol is just useless, so we must define an atol. 
If we want the tolerance to rely on the type, we must separate Float and Integer case (as the default rtol do), because eps is not defined for Integer types.
This is not an absurd way o do it as rtol is defined quite similarly:
```
# default tolerance arguments
rtoldefault(::Type{T}) where {T<:AbstractFloat} = sqrt(eps(T))
rtoldefault(::Type{<:Real}) = 0
function rtoldefault(x::Union{T,Type{T}}, y::Union{S,Type{S}}, atol::Real) where {T<:Number,S<:Number}
    rtol = max(rtoldefault(real(T)),rtoldefault(real(S)))
    return atol > 0 ? zero(rtol) : rtol
end
```
See [Base.isapprox](https://docs.julialang.org/en/v1/base/math/#Base.isapprox)
and [floatfuncs](https://github.com/JuliaLang/julia/blob/master/base/floatfuncs.jl) lines 285-291 for the definition of rtol

Reply:
I don't think so, `0.1+0.2-0.3` is much greater than subnormal numbers

Reply:
After experimentation, comparing to the epsilon is not sufficient for large float.
It seems sqrt(epsilon) is a common choice for tolerance, and it is justified by mathematical error analysis. 
Maybe an even better choice would be to choose the tolerance depending on the capacity matrix values. For example, `tol = sqrt(eps(maximum(capacity_matrix)` or `tol = sqrt(eps(sum(capacity_matrix)`